addons/delivery/models/stock_picking.py

```diff
@@ -190,6 +224,20 @@ def check_packages_are_identical(self):
                 raise UserError(_('You are shipping different packaging types in the same shipment.\nPackaging Types: %s' % package_names))
         return True
 
+    @api.multi
+    @api.returns('self', lambda value: value.id)
+    def message_post(self, **kwargs):
+        """ Overrides mail_thread message_post so that we can set delivery_mail_sent to true  of boolean field value to identify email 
+        is sent and make send confirmation email button normal from primary.
+        """
+        if self.env.context.get('mark_delivery_as_sent'):
+            for picking in self:
+                if picking.state == 'done':
+                    picking.delivery_mail_sent = True
+        return super(StockPicking, self).message_post(**kwargs)
```

Comment:
Sounds like a terrible idea for performances, please remove that 

Why cant " picking.delivery_mail_sent = True" be done in action_send_confirmation_email ?